stacks/product-jenkins/versions.tf

```diff
@@ -13,12 +13,12 @@ terraform {
 
     harbor = {
       source  = "liatrio/harbor"
-      version = "= 0.3.3"
+      version = "0.4.0"
```

Comment:
this actually fixes the delete jobs failing, so we can put off updating this provider until later